python/cudf/cudf/core/index.py

```diff
@@ -2703,6 +2706,144 @@ def categories(self):
         return self._values.cat().categories
 
 
+def interval_range(
+    start: int = None,
+    end: int = None,
+    periods: int = None,
+    freq: int = None,
+    closed: str = "right",
+    name: str = None,
+) -> "IntervalIndex":
+    """
+    Returns a fixed frequency IntervalIndex.
+
+    Parameters
+    start : numeric, default None
+    Left bound for generating intervals.
+
+    end : numeric , default None
+    Right bound for generating intervals.
+
+    freq : numeric, default None
+    The length of each interval. Must be consistent
+    with the type of start and end
+
+    name : str, default None
+    Name of the resulting IntervalIndex.
+
+    closed : {‘left’, ‘right’, ‘both’, ‘neither’}, default ‘right’
+    Whether the intervals are closed on the left-side, right-side,
+    both or neither.
+
+    Returns
+    IntervalIndex
+
+    Examples
+    --------
+    >>> import cudf
+    >>> import pandas as pd
+    >>> cudf.interval_range(start=0,end=5)
+    ... IntervalIndex([(0, 0], (1, 1], (2, 2], (3, 3], (4, 4], (5, 5]],
+              closed='right',dtype='interval')
+    """
+    if freq and periods and start and end:
+        raise ValueError(
+            "Of the four parameters: start, end, periods, and "
+            "freq, exactly three must be specified"
+        )
+    elif periods and not freq:
+        if end is not None:
+            end = end + 1
+        periods_array = cupy.asarray(cupy.arange(start, end))
+        hist, bin_edges = cupy.histogram(periods_array, periods)
```

Comment:
```suggestion
        _, bin_edges = cupy.histogram(periods_array, periods)
```

Reply:
Decided to go with another approach. Just running a few more tests and will update :D

Reply:
Ended up sticking with this! Added these changes :D Thank you!